### ISSUE TITLE
I2C devices overall don't work

### ISSUE THREAD

**Issue by koluris** [dkoluris/HP-ENVY-13-AH0002-OpenCore#3]
Even though investigating this laptop's DSDT shows us that there are two active I2C devices namely `TPD0` and `TPL1`, none of them seem to work with VoodooI2C current implementation.

If we make those work, it could mean that the trackpad could have better response, and also eventually make the touchscreen work.

Here's the repo: https://github.com/VoodooI2C/VoodooI2C

**Comment by rholelaw** [comment]
I've created a Pull Request with some changes and a DSDT to expose TPD0 and TPL1. Added VoodooI2C.  While digging into this problem with TPL1 I accidentally fixed it so the Sintek SDReader driver works.

My Envy is using the latest BIOS Firmware so make sure to be updated to not have issues with the DSDT not working.

**Comment by koluris** [comment]
Thank you so much for your contribution to this. I have actually investigated your PR and it looks great in most aspects. I just made the changes on top of my current DSDT because while using a diff merge tool I noticed some different hex numbers between mine and yours, I am guessing this is due to the fact that you have also a dGPU, while I don't.

Nonetheless, your contribution is awesome, and thanks a lot about it!
